openlibrary/plugins/openlibrary/js/ol.js

```diff
@@ -75,48 +79,263 @@ export function closePop(){
+        if (opts.fireAtStart) handler(this[key]);
+    }
+
+    /**
+     * @param {String} key
+     * @param {any} newValue
+     * @param {any} oldValue
+     */
+    _trigger(key, newValue, oldValue) {
+        if (!(key in this._listeners)) {
+            return;
+        }
+
+        for (let listener of this._listeners[key]) {
+            if (listener.opts.onlyFireOnChange) {
+                if (newValue != oldValue) {
+                    listener.handle(newValue)
+                }
+            } else {
+                listener.handle(newValue);
+            }
+        }
+    }
+}
+
+class SearchBar {
+    /**
+     *
+     * @param {SearchState} searchState
+     * @param {Object} urlParams
+     */
+    constructor(searchState, urlParams) {
+        this.searchState = searchState;
+        /** The search input element */
+        this.$searchInput = $('header#header-bar .search-component .search-bar-input input[type="text"]');
+        /** Autocomplete search results */
+        this.$searchResults = $('header#header-bar .search-component ul.search-results');
+        /** stores the state of the search result for resizing window */
+        this.instantSearchResultState = false;
+        /** Whether the search bar is expanded */
+        this.searchExpansionActivated = false;
+        /** ?? Not sure */
+        this.enteredSearchMinimized = false;
+
+        if (urlParams.q) {
+            let q = urlParams.q.replace(/\+/g, ' ');
+            if (searchState.facet === 'title' && q.indexOf('title:') != -1) {
+                const parts = q.split('"');
+                if (parts.length === 3) {
+                    q = parts[1];
+                }
+            }
+            $('.search-bar-input [type=text]').val(q);
+        }
+
+        if ($(window).width() < 568) {
+            if (!this.enteredSearchMinimized) {
+                $('.search-bar-input').addClass('trigger')
+            }
+            this.enteredSearchMinimized = true;
+        }
 
-    // Search mode
-    searchModes = ['everything', 'ebooks', 'printdisabled'];
-    searchModeDefault = 'ebooks';
-
-    // Maps search facet label with value
-    defaultFacet = 'all';
-    searchFacets = {
-        'title': 'books',
-        'author': 'authors',
-        'lists': 'lists',
-        'subject': 'subjects',
-        'all': 'all',
-        'advanced': 'advancedsearch',
-        'text': 'inside'
-    };
-
-    composeSearchUrl = function(q, json, limit) {
-        var facet_value = searchFacets[localStorage.getItem('facet')];
-        var url = ((facet_value === 'books' || facet_value === 'all')? '/search' : `/search/${facet_value}`);
+        // searches should be cancelled if you click anywhere in the page
+        $(document.body).on('click', this.cancelSearch.bind(this));
+        // but clicking search input should not empty search results.
+        $(window).resize(this.handleResize.bind(this));
+        this.$searchInput.on('click', false);
```

Comment:
I don't think this should mix responsibilities of adding JS events as well as localstorage.

Reply:
See the later iterations with `PersistentValue`